### ISSUE TITLE
bug with animation start/done events in unit tests in 4.2.0

### ISSUE THREAD

**Issue by emoralesb05** [angular/angular#17375]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When using a trigger animation such as a `fade` animation with opposing states:

```
trigger('fade', [
    state('0', style({
      opacity: '0',
    })),
    state('1',  style({
      opacity: '*',
    })),
    transition('0 => 1', animate(duration + 'ms ease-in')),
    transition('1 => 0', animate(duration + 'ms ease-out')),
  ]);
```

binding to its start and done events works fine when wanting to execute something. 

The problem is while unit testing, when going from `0 => 1` triggers the events in the following order `@fade.start => @fade.end`.. but for some reason when going from state `1 => 0` it triggers them in reverse order `@fade.end => @fade.start`

**Expected behavior**
I would expect the order of the events to always be `@fade.start => @fade.end` regardless of the state that triggered it.

**Minimal reproduction of the problem with instructions**
To replicate this you need to unit test an animation with opposite states like booleans or something like that.

**What is the motivation / use case for changing the behavior?**
Behavior in unit test is different from actual behavior.

**Please tell us about your environment:**
Happens in any env, tried it in travis and mac.

* **Angular version:** 2.0.X
4.2.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Unit testing in chrome.
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript 2.2

**Comment by emoralesb05** [comment]
I was hooking into the events via `@HostListener` if it helps

**Comment by matsko** [comment]
@emoralesb05 there have been some fixes in the 5 or so releases since 4.2.0. Could you retest your code against 4.2.5 and see if the issue still exists?

**Comment by vicb** [comment]
Closing as no repro has been provided since Jun 17
